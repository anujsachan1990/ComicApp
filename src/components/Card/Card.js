import React, { memo } from 'react'
import { StyleSheet, View, Text } from 'react-native'
import theme from '../../styles/theme'
import locale from '../../lang/en-us'

const Card = ({ item }) => {
  return (
    <View key={item.diamond_id} style={styles.content}>
      <Text style={styles.largeText}>{item.title}</Text>
      <Text style={styles.smallText}>
        <Text style={styles.boldText}>{locale.description} </Text>
        {item.description}
      </Text>
      <Text style={styles.smallText}>
        <Text style={styles.boldText}>{locale.price} </Text>
        {item.price}
      </Text>
      <Text style={styles.smallText}>
        <Text style={styles.boldText}>{locale.publisher}</Text>
        {item.publisher}
      </Text>
      <Text style={styles.smallText}>
        <Text style={styles.boldText}>{locale.relaseData}</Text>
        {item.release_date}
      </Text>
      <Text style={styles.smallText}>
        <Text style={styles.boldText}>{locale.creators}</Text> {item.creators}
      </Text>
    </View>
  )
}
export default memo(Card)

const styles = StyleSheet.create({
  content: {
    backgroundColor: theme.color.primaryColor,
    padding: 20,
    margin: 20,
    borderRadius: 10,
    minHeight: 300,
  },
  smallText: {
    fontSize: theme.font.bodycopy,
    marginTop: 10,
    color: theme.color.white,
  },
  largeText: {
    fontSize: theme.font.heading,
    color: theme.color.white,
  },
  boldText: {
    fontWeight: 'bold',
  },
})
