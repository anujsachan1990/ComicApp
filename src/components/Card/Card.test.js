import React from 'react'
import renderer from 'react-test-renderer'

import Card from './Card'

const props = {
  item: {
    title: 'Title',
    price: 'price',
    description: 'description',
    publisher: 'publisher',
    release_date: 'release_date',
    creators: 'creators',
  },
}

describe('<Card />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Card {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
