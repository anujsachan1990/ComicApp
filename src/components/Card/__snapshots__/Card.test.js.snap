// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#00838d",
      "borderRadius": 10,
      "margin": 20,
      "minHeight": 300,
      "padding": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 20,
      }
    }
  >
    Title
  </Text>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 15,
        "marginTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Description: 
       
    </Text>
    description
  </Text>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 15,
        "marginTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Price: 
       
    </Text>
    price
  </Text>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 15,
        "marginTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Publisher :
    </Text>
    publisher
  </Text>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 15,
        "marginTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Release date :
    </Text>
    release_date
  </Text>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 15,
        "marginTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Creators :
    </Text>
     
    creators
  </Text>
</View>
`;
